libs/experimental/langchain_experimental/autonomous_agents/autogpt/agent.py

```diff
@@ -129,15 +127,39 @@ def run(self, goals: List[str]) -> str:
                     f"commands and only respond in the specified JSON format."
                 )
 
-            memory_to_add = (
-                f"Assistant Reply: {assistant_reply} " f"\nResult: {result} "
-            )
+            # show result so user actually know their answer
+            print(result)
+
+            self.chat_history_memory.add_message(HumanMessage(content=user_input))
+            self.chat_history_memory.add_message(AIMessage(content=assistant_reply))
+            self.chat_history_memory.add_message(SystemMessage(content=result))
```

Comment:
this is an incorrect use of SystemMessage. Maybe include in the AIMessage or as a second AIMessage?